docs/model_developers/modules.rst

```diff
@@ -329,3 +329,41 @@ for supplying the agents with the appropriate ``inmanta_plugins`` packages.
 The only exception to this rule is when using the ``inmanta export`` command. It exports a project and all its modules'
 ``inmanta_plugins`` packages to the orchestrator server. When this method is used, the orchestrator does not install any modules
 from the Python package repository but instead contains all Python code as present in the local Python environment.
+
+Configure a project to install modules form a private python package repository
+--------------------------------------------------------------------------------
+
+Modules v2 are Python packages that can be installed from private Python package repositories, which may require authentication. This section explains the steps to install such modules.
+
+Create a file named ``.netrc`` in the orchestrator's file system, specifically in the "inmanta" directory. 
+Add the following content to the file:
+
+.. code-block:: text
+
+  machine <url of the private repository>
+  login token
+  password <the token to access the private repository>
+
+In the ``project.yml`` file of your project, ensure that the ``use_config_file`` option under the ``pip`` section is set to ``True``.
+
+Create a file named ``pip.conf`` and add the following content to it:
+
+.. code-block:: text
+
+  [global]
+  timeout = 60
+  index-url = <url of the repository >
+
+Set the ``PIP_CONFIG_FILE`` environment variable to the path of the ``pip.conf`` file. You can do this by running the following command:
```

Comment:
I am not sure whether we want to create a new pip config file here and set the environment variable or whether it would be better to change the global pip config file. I guess the latter is better, but I am not sure.

Reply:
I could also speak about booth options as they both work?

Reply:
That's a possibility. Regarding the approach with the environment variable: I would mention the procedure to make sure that the environment variable is automatically set when a bash shell is opened. Having to run the export command every time a user logs into the orchestrator is not a very practical solution.

Reply:
but this part should be removed completely right? or do I misunderstand [your other comment](https://github.com/inmanta/inmanta-core/pull/6194#discussion_r1245274243)?

Reply:
I don't really see how the referenced comment is linked to this. If we use the global pip config instead of a custom config file, this can indeed be removed.